src/replication.c

```diff
@@ -151,14 +152,18 @@ perform_read_write_on_fd(int fd, uint8_t *data, uint64_t len, int state)
 		}
 
 		nbytes += rc;
-		if(nbytes == len) {
+		if((size_t)nbytes == len) {
 			break;
 		}
 	}
 
 	return nbytes;
 }
 
+/*
+ * get_all_read_resp_data_chunk will create/update io_data_chunk_list
+ * with response received from replica according to io_numer.
+ */
 void
 get_all_read_resp_data_chunk(replica_rcomm_resp_t *resp, struct io_data_chunk_list_t *io_chunk_list)
```

Comment:
this needs fix in case of all degraded replicas scenario. if I'm not wrong, this will be handled in next PRs.

Reply:
yes